File: src/main/java/com/bjsxt/sparkjavacode/SparkCache.java
package com.bjsxt.sparkjavacode;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.storage.StorageLevel;

public class SparkCache {
    public static void main(String[] args) {
        JavaSparkContext sc = new JavaSparkContext(new SparkConf().setMaster("local").setAppName("test"));

        JavaRDD<String> lines = sc.textFile("./data/words");

        JavaRDD<String> persist = lines.persist(StorageLevel.MEMORY_ONLY());

        long timeMillis = System.currentTimeMillis();
        long count = persist.count();
        long endtimeMillis = System.currentTimeMillis();
    }
}

File: src/main/scala/com/bjsxt/sparkscalacode/SparkCache.scala
package com.bjsxt.sparkscalacode

import org.apache.spark.{SparkConf, SparkContext}

object SparkCache {
  def main(args: Array[String]): Unit = {
    val sc = new SparkContext(new SparkConf().setMaster("local").setAppName("test"))

    val lines = sc.textFile("./data/persistData.txt")

    lines.cache()

    val startTime1 = System.currentTimeMillis()
    val count1 = lines.count()
    val startTime2 = System.currentTimeMillis()
  }
}
